plugins/retry429.js

```diff
@@ -0,0 +1,44 @@
+// Copyright © 2017 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+const BasePlugin = require('./base.js');
+
+/**
+ * Retry 429 response plugin.
+ *
+ * @param {any} client - HTTP client.
+ * @param {any} opts - Plugin options.
+ */
+function Retry429Plugin(client, opts) {
+  Retry429Plugin.super_.apply(this, arguments);
```

Comment:
Any reason this approach is preferred over the ES2105 `class`/`constructor` approach? Personally I find it easier to follow, and it is pretty widely supported now. Although I guess the Node engine `0.8.0` which the `package.json` declares as a minimum is too early for that. I think we only test on the LTS 4, LTS 6 and current now, so maybe we can update that minimum engine anyway.